### DISCUSSION TITLE
Query on an Oracle database with AS format problem

### DISCUSSION THREAD

**Discussion by micgerber** [sqlalchemy/sqlalchemy#9769] [Usage Questions] [answered]
Hi,

I'm using FastAPI to execute SQLAlchemy query on an Oracle Cloud Autonomous Database (v. 19c).

I have the following query in my crud.py file : 

```python
from sqlalchemy.orm import Session 
import models, schemas

def get_users(db: Session, skip: int = 0, limit: int = 100):
    return db.query(models.User).offset(skip).limit(limit).all()
```

and the following class in my models.py file : 

```python
from sqlalchemy import Boolean, Column, ForeignKey, Integer, MetaData, String, Table
from sqlalchemy.orm import relationship
from database import SessionLocal, engine

from database import Base

class User(Base):
    metadata = MetaData()
    __tablename__ = "PIC.user"
    __table_args__ = {'quote': False}

    id = Column(Integer, primary_key=True, index=True, nullable=False)
    username = Column(String, nullable=True)
```

I am accessing the database with the user PIC_APP who have access to the table PIC.user (the table user of the Oracle user PIC).

I have the following error : 

```
sqlalchemy.exc.DatabaseError: (cx_Oracle.DatabaseError) ORA-00923: FROM keyword not found where expected
[SQL: SELECT PIC.user.id AS PIC.user_id, PIC.user.username AS PIC.user_username
FROM PIC.user
```

I need the query to be one of the following forms : 

```sql
SELECT PIC.user.id AS "PIC.user_id", PIC.user.username AS "PIC.user_username" FROM PIC.user
```

or 

```sql
SELECT PIC.user.id AS user_id, PIC.user.username AS user_username FROM PIC.user
```

or 

```sql
SELECT PIC.user.id, PIC.user.username FROM PIC.user
```

because the AS format with a dot and no quote doesn't work.

By the way, if I remove : 

```python
__table_args__ = {'quote': False}
```
this also doesn't work.

**Comment by CaselIT** [comment]
Hi,

if the table is not named `"PIC.user"`, but `"PIC"` is the schema, it should be in the schema, not on the name of the table. Try using `__table_args__ = {"schema": "PIC"}` and `__tablename__ = "user"`

**Reply by micgerber** [reply]
Thank you very much for your answer.
This works perfectly.

**Comment by zzzeek** [comment, marked as answer]
use the [schema](https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#orm-declarative-table-schema-name) parameter and never change quoting.  if the name "PIC" is case insensitive in Oracle, use all lower case to indicate case-insensitive:


```py
class User(Base):
    metadata = MetaData()
    __tablename__ = "user"
    __table_args__ = {'schema': "pic"}

    id = Column(Integer, primary_key=True, index=True, nullable=False)
    username = Column(String, nullable=True)
```

**Reply by micgerber** [reply]
Thank you very much for your DETAILED answer.
This works perfectly.
